scrnaseq.Rmd

```diff
@@ -1761,8 +1768,56 @@ if (markers_found) {
 }
 ```
 
-### Heatmaps of differentially expressed genes {.tabset}
-#### All up- and down-regulated genes
+## Dotplots per cluster per sample 
+If the dataset contains multiple samples, we can visualise the expression of a gene that is up-regulated in a cluster separately for each sample. For each cluster, we extract up-regulated genes, and visualise expression of these genes in all cells in that cluster, split by their sample of origin. 
+```{r degs_plot_dotplotpercl_height}
+fig_height_degs_per_cl = max(5, 
+                             max(2, 0.3 * (sc$orig.ident %>% unique() %>% length())) * length(levels(sc$seurat_clusters)) * 2)
+```
+
+First, we plot scaled expression as explained above (see section Known marker genes). This plot allows us to judge whether the expression of a gene is increased in one sample as compared to the other samples.
+```{r degs_plot_dotplotpercl_scaled, fig.height=fig_height_degs_per_cl}
+n_genes_max_dotplot = 50
+p_list = list()
+for (cl in levels(sc$seurat_clusters)) {            
+  markers_filt_up_cl_top = markers_filt$up %>% 
+    dplyr::filter(cluster==cl) %>% 
+    dplyr::top_n(n=n_genes_max_dotplot, wt=p_val_adj_score) %>% 
+    dplyr::pull(gene)
+  if (length(markers_filt_up_cl_top) > 0) {
+    p_list[[cl]] = suppressMessages(Seurat::DotPlot(sc, features=markers_filt_up_cl_top, idents=cl, group.by="orig.ident") +
```

Comment:
For now drop the `group.by="orig.ident"` ?

Reply:
No, I am not going to mess with the dotplots again. This took me time to understand and adapt. Why do you want me to drop this?

Reply:
Hmm, I had a look at the html and the dot plots look fine. But can you document/explain why do you need `group.by="orig.ident" ` here? That is unclear here

Reply:
Ok, I'll document it better.

Reply:
I added this piece of documentation, and hope it makes it more clear: 

```
# Explanation Seurat::DotPlot parameters
# By default, cells are divided into groups (= rows in the dotplot) based on their default identity
# Parameter idents: We can sub-select cells that we want to include in the plot
# Parameter group.by: We can define ourselves, based on what factor we want to group cells 
# Parameter split.by: We could split the cells even further based on another factor, and assign a different colour for each split
# That is, here we select only cells that fall into one cluster (idents=cl), and group those based on their sample of origin (group.by="orig.ident")
```